src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,96 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+Base.transpose(::AbstractDataFrame, args...; kwargs...) =
+    MethodError("`transpose` not defined for `AbstractDataFrame`s. Try `permutedims` instead")
+    # ↑ is 94 char, is that OK? (there are a couple other lines in this file that go over)
+
+"""
+    permutedims(df::AbstractDataFrame, src_namescol::ColumnIndex,
+                dest_namescol::Union{Symbol,Int};
+                makeunique::Bool=false)
+    permutedims(df::AbstractDataFrame, src_namescol::ColumnIndex; makeunique::Bool=false)
+    permutedims(df::AbstractDataFrame; makeunique::Bool=false)
+
+Turn a `DataFrame` on its side such that rows become columns
+and the column indexed by `src_namescol` becomes a header.
+In the resulting `DataFrame`,
+The header of `df` will become the first column
+with name specified by `dest_namescol`
+
+# Arguments
+- `df` : the `AbstractDataFrame`
+- `src_namescol` : the column that will become the new header.
+  Defaults to first column.
+- `dest_namescol` : the name of the first column in the returned `DataFrame`.
+  Defaults to the same name as `src_namescol`.
+- `makeunique` : if `false` (the default), an error will be raised
+  if duplicate names are found; if `true`, duplicate names will be suffixed
+  with `_i` (`i` starting at 1 for the first duplicate).
+
+Note: The eltypes of columns in resulting `DataFrame` will depend
+on the eltypes of _all_ input columns.
+That is, if the source `DataFrame` contains `String` and `Int` columns,
+all resulting columns will have eltype `Any`.
+If the source has a mix of numeric types (eg. `Float64` and `Int`),
+all columns in resulting `DataFrame` will be promoted to `Float64`.
+
+# Examples
+
+```jldoctest
+julia> df1 = DataFrame(a=["x", "y"], b=rand(2), c=[1,2], d=rand(Bool,2));
```

Comment:
do not use random values (I know we used them in the past in some places, but it is not reproducible).
Also I think showing the source data frames is in this case desirable to visualize the difference.